### PR TITLE
Material2 alpha.5 and tabs

### PR DESCRIPTION
**Pull request by justindujardin**
- Update to latest material2 alpha and deprecate tabs in favor of official implementation. 
- Update existing tabs component examples and view source docs-example component.

### CODE REVIEW

File: modules/site/src/app/examples/button/button-basic-usage.component.spec.ts [hunk 1]
```diff
@@ -1,4 +1,11 @@
-import {beforeEach, beforeEachProviders, describe, expect, it, inject,} from '@angular/core/testing';
+import {
+  beforeEach,
+  beforeEachProviders,
+  describe,
+  expect,
+  it,
+  inject,
```
Comment (Gregcop1) [inline modules/site/src/app/examples/button/button-basic-usage.component.spec.ts]: useless comma (same for other files)

File: modules/site/.gitignore [hunk 2]
```diff
@@ -25,4 +25,7 @@ testem.log
 /e2e/*.map
 
 #System Files
-.DS_Store
\ No newline at end of file
+.DS_Store
+
+# Meta files generated by ng2-material build
+public/*.json
```
Comment (Gregcop1) [inline modules/site/.gitignore]: IMO, it should be in a hotfix cause it's not directly related to this PR and it concern several pending pull requests.

File: modules/site/src/app/examples/button/button-basic-usage.component.spec.ts (hunk 1 above)
Comment (justindujardin) [inline modules/site/src/app/examples/button/button-basic-usage.component.spec.ts]: That's unfortunate.  I just ran the latest `ng format` on the examples app and it did this.  Can we override the formatter styles?  I tend to agree that trailing commas are not helpful.

Comment (Gregcop1) [inline modules/site/src/app/examples/button/button-basic-usage.component.spec.ts]: On my project I bannished usage of `ng format` :/

File: modules/site/.gitignore (hunk 2 above)
Comment (justindujardin) [inline modules/site/.gitignore]: Done, cherry picked the removal into master with your pagination PR.

**Comment by justindujardin** [comment]
* feat(material2): update to alpha 5-2

**Comment by justindujardin** [comment]
I'm going to go ahead and merge this because if I don't now it may be a while until I get back to it. I know the coverage dropped, but that's mostly due to me removing the _spec.ts files that were being counted before.

File: modules/site/src/app/examples/button/button-basic-usage.component.spec.ts (hunk 1 above)
Comment (justindujardin) [inline modules/site/src/app/examples/button/button-basic-usage.component.spec.ts]: I'd prefer to figure out how to tweak the style it uses. It's really nice to make sure code is all stylistically consistent. It's been my experience that even professional developers have a hard time being consistent with each other if you don't have a build process in the middle double-checking it.

Comment (ollwenjones) [inline modules/site/src/app/examples/button/button-basic-usage.component.spec.ts]: Add [codelyzer](https://github.com/mgechev/codelyzer) to a `tslint` process? Gives developers warnings about formatting inconsistencies, and lets them decide how to make their files match it, (versus a script that does things like add trailing commas, break lines at harder-to-read places, etc.)

**Pull request merged**